anxcloud/utils.go

```diff
@@ -58,3 +58,22 @@ func fieldsComputed(fields ...string) func(schemaMap) {
 		}, fields...)
 	}
 }
+
+func listAllPages[T any](pageRetriever func(int) ([]T, error)) ([]T, error) {
+	var allPages = make([]T, 0)
+
+	for page := 1; true; page++ {
+		singlePage, err := pageRetriever(page)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(singlePage) == 0 {
+			break
```

Comment:
brave. I know at least one API that returns the same data on every page :upside_down_face: 

but since this is added manually per API, it's probably fine

ref:
* https://github.com/anexia-it/go-anxcloud/pull/69
* https://github.com/anexia-it/go-anxcloud/pull/82